ecommerce/frontend/src/components/SearchInput.js

```diff
@@ -0,0 +1,53 @@
+import React, { useState } from 'react'
+import './SearchInput.less'
+import { Select, Input } from 'antd'
+
+const { Option } = Select
+const { Search } = Input
+
+// TODO - Fetch from a common file or backend
+const categories = [
+    { key: 'electronics', title: 'Electronics' },
+    { key: 'home_and_garden', title: 'Home & Garden' },
+    { key: 'personal_care', title: 'Personal Care' },
+    { key: 'furniture', title: 'Furniture' },
+    { key: 'pet_supplies', title: 'Pet Supplies' },
+    { key: 'mens_fashion', title: "Men's Fashion" },
+    { key: 'womens_fashion', title: "Women's Fashion" },
+]
+
+export const SearchInput = (width = 120, onSearch) => {
+    const [category, setCategory] = useState('Electronics')
+
+    function handleCategoryChange(value) {
+        setCategory(value)
+    }
+
+    const categoryDropdown = (
+        <Select defaultValue={category} style={{ width: { width } }} onChange={handleCategoryChange}>
+            {categories.map(category => (
+                <Option value={category.title} key={category.key}>
+                    {category.title}
+                </Option>
+            ))}
+        </Select>
+    )
+
+    const onSearchInput = value => {}
```

Comment:
I am not sure about the purpose of  `onSearchInput` ? Asking only because I am curious. I don't have anything more to say.

Reply:
`onSearchInput` will work for filtering the products according to the input value.

We may have a container, say "PRODUCTS", which will make a call to Backend and get all the products. Then, we display them in the PRODUCTS container. 

When a customer enters input and clicks the search icon ->  `onSearchInput` will raise an event to PRODUCTS to apply the filtering, according to the input.

Reply:
@abdullahyldz I think I understand what you are trying to say, but the `onSearchInput` is not referred to in any place in the code? and aren't you already using `onSearch` prop to tell the parent component that a search was performed? I think onSearchInput is not used here.

Reply:
Guys, I am sorry. I mixed `onSearchInput `with `SearchInput `and wrote about the component :) 
I moved the calls inside `onSearchInput ` to inside of Search component and I forgot to delete `onSearchInput `
 ```
         <Search
                placeholder="Search for products, categories and brands"
                onSearch={value => {
                    if (value.trim() === '') {
                        return
                    }
                    onSearch(value)
                }}
                enterButton
         />
```

Reply:
I updated my comment above to a more correct version